android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java

```diff
@@ -519,15 +523,30 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent keyEvent)
 			return true;
 		}
 
-		//This is to prevent 'return' event from being fired twice when return key is hit. In other words, when return key is clicked,
-		//this callback is triggered twice (except for keys that are mapped to EditorInfo.IME_ACTION_NEXT or EditorInfo.IME_ACTION_DONE). The first check is to deal with those keys - filter out
-		//one of the two callbacks, and the next checks deal with 'Next' and 'Done' callbacks, respectively.
-		//Refer to TiUIText.handleReturnKeyType(int) for a list of return keys that are mapped to EditorInfo.IME_ACTION_NEXT and EditorInfo.IME_ACTION_DONE.
-		if (actionId == EditorInfo.IME_ACTION_NEXT || actionId == EditorInfo.IME_ACTION_DONE) {
+		data.put(TiC.PROPERTY_BUTTON, RETURN_KEY_TYPE_ACTION);
+		// Check whether we a dealing with text area or text field. Multiline TextViews in Landscape
+		// orientation for phones have separate buttons for IME_ACTION and cartridge return.
+		// And because of that we skip the firing of a RETURN event from this call in favor of the
+		// one from onTextChanged. The event carries a property to determine whether it was fired
+		// from the IME_ACTION button or the cartridge return one.
+		if (tv.getMaxLines() == 1) {
```

Comment:
Instead of checking if `getMaxLines()` is 1, we should use the `field` member variable instead. It'll be set to `true` for TextFields.

Note that upcoming PR #9927 will make max-lines settable and I don't want you to get burned by this.